### PR TITLE
Move 'export name' out of the import section.

### PR DESCRIPTION
**Pull request by jonmeow**
This implements recent updates to #3938

### CODE REVIEW

**Comment by jonmeow** [comment]
Depends on #3949

File: toolchain/parse/state.def [hunk 1]
```diff
@@ -670,35 +670,31 @@ CARBON_PARSE_STATE(FunctionSignatureFinish)
...
 
-// Handles `export`.
-//
-// export import library "libname";
-// ^~~~~~
-//   1. Import
+// Handles `export <name>`.
 //
 // export Name;
 // ^~~~~~
```
Comment (geoffromer) [inline toolchain/parse/state.def]: It looks like `HandleExportName` expects the `export` to already have been consumed by `HandleDeclScopeLoop`, so should this `^~~~~~` be moved to the comment on `DeclScopeLoop`?

Comment (jonmeow) [reply toolchain/parse/state.def]: Done

**Review by geoffromer — APPROVED** [review]

**Comment by chandlerc** [comment]
(Removed from merge queue to let #3949 finish merging as that's a dependency of this PR)

**Pull request merged**